src/agent/onefuzz-supervisor/src/main.rs

```diff
@@ -123,13 +129,20 @@ async fn run_agent(config: StaticConfig) -> Result<()> {
     telemetry::set_property(EventData::InstanceId(config.instance_id));
     telemetry::set_property(EventData::MachineId(get_machine_id().await?));
     telemetry::set_property(EventData::Version(env!("ONEFUZZ_VERSION").to_string()));
-    if let Ok(scaleset) = get_scaleset_name().await {
-        telemetry::set_property(EventData::ScalesetId(scaleset));
+    let scaleset_result = get_scaleset_name().await;
```

Comment:
Instead of using the `Result` variant to signal info about our environment, maybe `get_scaleset_name()` should return a `Result<Option<String>>`.

Even better, you could introduce a new function that returns a enum like:
```rust
enum NodeType {
    Managed { scaleset_name: String },
    Unmanaged,
}

impl NodeType {
    pub fn get() -> Result<Self> { ... }
}
```

Reply:
Added the former.